io.c

```diff
@@ -10922,62 +10918,82 @@ pipe_pair_close(VALUE rw)
- *     IO.pipe(ext_enc)                    ->  [read_io, write_io]
- *     IO.pipe("ext_enc:int_enc" [, opt])  ->  [read_io, write_io]
- *     IO.pipe(ext_enc, int_enc [, opt])   ->  [read_io, write_io]
+ *    IO.pipe(enc_string = 'UTF-8', **opts) -> [read_io, write_io]
+ *    IO.pipe(enc_string = 'UTF-8', **opts) {|read_io, write_io] ...} -> object
+ *    IO.pipe(int_enc = Encoding::UTF_8, ext_enc = nil, **opts) -> [read_io, write_io]
+ *    IO.pipe(int_enc = Encoding::UTF_8, ext_enc = nil, **opts) {|read_io, write_io] ...} -> object
  *
- *     IO.pipe(...) {|read_io, write_io| ... }
+ *  Creates a pair of pipe endpoints, +read_io+ and +write_io+,
+ *  connected to each other.
  *
- *  Creates a pair of pipe endpoints (connected to each other) and
- *  returns them as a two-element array of IO objects:
- *  <code>[</code> <i>read_io</i>, <i>write_io</i> <code>]</code>.
+ *  If argument +enc_string+ is given, it must be a string containing one of:
  *
- *  If a block is given, the block is called and
- *  returns the value of the block.
- *  <i>read_io</i> and <i>write_io</i> are sent to the block as arguments.
- *  If read_io and write_io are not closed when the block exits, they are closed.
- *  i.e. closing read_io and/or write_io doesn't cause an error.
+ *  - The name of the encoding to be used as the internal encoding.
+ *  - The colon-separated names of two encodings to be used as the internal
+ *    and external encodings.
  *
- *  Not available on all platforms.
+ *  An array of the encoding names is returned by method Encoding.name_list.
+ *
+ *  If argument +int_enc+ is given, it must be an encoding object
+ *  that specifies the internal encoding to be used;
+ *  if argument +ext_enc+ is also given, it must be an encoding object
+ *  that specifies the external encoding to be used.
+ *
+ *  An array of encoding classes is returned by Encoding.list.
+ *
+ *  The string read from +read_io+ is tagged with the internal encoding;
+ *  if an external encoding is also specified, the string is converted
+ *  to, and tagged with, that encoding.
  *
- *  If an encoding (encoding name or encoding object) is specified as an optional argument,
- *  read string from pipe is tagged with the encoding specified.
- *  If the argument is a colon separated two encoding names "A:B",
- *  the read string is converted from encoding A (external encoding)
- *  to encoding B (internal encoding), then tagged with B.
- *  If two optional arguments are specified, those must be
- *  encoding objects or encoding names,
- *  and the first one is the external encoding,
- *  and the second one is the internal encoding.
  *  If the external encoding and the internal encoding is specified,
  *  optional hash argument specify the conversion option.
  *
+ *  Optional argument +opts+ must specify valid open options
+ *  (see {IO Open Options}[#class-IO-label-Open+Options])
+ *  and/or valid encoding options (see String#encode).
```

Comment:
Maybe this too requires examples... Or I am just overthinking stuff :joy:

Reply:
This is awkward.  I haven't yet given the simple examples, so am reluctant to offer a more complex one.  Maybe a couple of examples at Open Options, to show how the options look when added?

Reply:
Let's disregard that for now. 
If the current docs are clear, _adding_ some examples can be postponed to the future.